addons/api/addon/serializers/storage-bucket.js

```diff
@@ -72,8 +72,17 @@ export default class StorageBucketSerializer extends ApplicationSerializer {
     const value = super.serializeAttribute(...arguments);
     const { credentialType } = snapshot.record;
     const { options } = attribute;
+
     // This deletes any fields that don't belong to the record's credential type
     if (options.isNestedAttribute && json.attributes) {
+      if (json.attributes?.region) {
+        json.attributes.region = json.attributes.region.trim();
+      }
+
+      if (credentialType == 'dynamic' && json.attributes?.role_arn) {
+        json.attributes.role_arn = json.attributes.role_arn.trim();
+      }
```

Comment:
Rather than manually writing code to trim each field which might have to be done for multiple resources (this ticket also included host sets), I think we should instead make this a bit more generic and customize our application serializer to handle this since it looks like it will be a common scenario.

You could first add a new field to each attribute on models that need it which tells us we should trim the whitespace, something like `trimWhitespace: true`. Then customize the [serializeAttribute](https://api.emberjs.com/ember-data/4.12/classes/RESTSerializer/methods/serializeAttribute?anchor=serializeAttribute) in our application serializer to handle this new case and that way you only have to annotate the attributes on the model for this behavior rather than having to customize each individual serializer that might need whitespace trimmed.

Reply:
I moved this check to the application level to solve the bug for storage buckets. The host set attr `preferred_endpoints` is a string-array so it will need a separate solution. @lisbet-alvarez is going to be on-call and will add this to her queue, if time allows, she will finish the solution.